proposals/memory-specializations/utilities.tex

```diff
@@ -0,0 +1,349 @@
+             @\newtxt{MoveConstructible<value_type_t<O>{>}()}@
+  @\changed{pair<InputIterator, OutputIterator>}{tagged_pair<tag::in(I), tag::out(O)>}@
+    uninitialized_move_n(@\changed{InputIterator}{I}@&& first, @\changed{Size}{difference_type_t<I>}@ n, @\changed{ForwardIterator}{O}@&& out)
+\end{codeblock}
+
+
+\pnum
+\effects Equivalent to:
+\begin{codeblock}
+        for (; n > 0; ++result, (void) ++first, --n)
+          ::new (static_cast<void*>(addressof(*result)))
+            @\changed{typename iterator_traits<ForwardIterator>::value_type}{value_type_t<O>}@(std::move(*first));
+        return {first,result};
+\end{codeblock}
+
+\pnum
+\remarks If an exception is thrown, some objects in \tcode{[first, next(first, n))} are left in a valid, but unspecified state.
+
+{\color{addclr}
+\pnum
+\requires \tcode{result} shall not be in the range \tcode{[first, next(first, n))}.
+} %%\color{addclr}
+
+\rSec3[uninitialized.fill]{uninitialized_fill}
+\begin{codeblock}
+  template <@\changed{class}{CopyConstructible}@ @\changed{ForwardIterator}{T}@, @\changed{class}{OutputIterator<const T\&>}@ @\changed{T}{O},@
+            @\added{Sentinel<O> S}@>
+    @\added{requires Same<value_type_t<T>\&, reference_t<T>{>}}@
+    @\changed{void}{O}@ fill(@\changed{ForwardIterator}{O}@ first, @\changed{ForwardIterator}{S}@ last, const T& x);
+\end{codeblock}
+
+\setcounter{Paras}{0}
+\pnum
+\effects \changed{As if by}{Equivalent to}:
+\begin{codeblock}
+        for (; first != last; ++first)
+           ::new (static_cast<void*>(addressof(*first)))
+              @\changed{typename iterator_traits<ForwardIterator>::value_type}{value_type_t<O>}@(x);
+        return last;
+\end{codeblock}
+{\color{addclr}
+\begin{codeblock}
+  template <CopyConstructible T, OutputRange<const T&> Rng>
+    requires Same<value_type_t<T>&, reference_t<T>>
+  safe_iterator_t<Rng>
+    fill(Rng&& rng, const T& x);
+\end{codeblock}
+
+\pnum
+\effects Equivalent to:
+\begin{codeblock}
+        return uninitialized_fill(begin(rng), end(rng), x);
+\end{codeblock}
+} %% \color{addclr}
+
+\begin{codeblock}
+  template <@\changed{class}{CopyConstructible}@ @\changed{ForwardIterator}{T}@, @\changed{class Size, class}{OutputIterator<const T\&>}@ @\changed{T}{O}@>
+    @\added{requires Same<value_type_t<T>\&, reference_t<T>{>}}@
+  @\changed{ForwardIterator}{O}@ fill_n(@\changed{ForwardIterator}{O}@ first, @\changed{Size}{difference_type_t<O>}@, const T& x);
+\end{codeblock}
+
+\setcounter{Paras}{0}
+\pnum
+\effects \changed{As if by}{Equivalent to}:
+\begin{codeblock}
+        for (; n > 0; --n, ++first)
+          ::new (static_cast<void*>(addressof(*first)))
+            @\changed{typename iterator_traits<ForwardIterator>::value_type}{value_type_t<O>}@(x);
+        @\added{return first;}@
+\end{codeblock}
+
+\rSec3[specialized.destroy]{destroy}
+
+\begin{codeblock}
+  template <@\changed{class}{ForwardIterator}@ @\changed{ForwardIterator}{I, Sentinel<I> S}@>
```

Comment:
Standard C++ needs `ForwardIterator` here, not for multipass, but to ensure that `I`'s reference type is a true reference. We can relax the category to input, but we need to ensure the reference property directly with e.g. 
`__ReferenceTo<I, value_type_t<I>>()`.